_meeting-notes/2019-12-03.md

```diff
@@ -0,0 +1,166 @@
+- **Participants:**
+    - @daviddias
+    - @nonsense
+    - @raulk
+    - @jimpick
+    - @momack2
+    - @stebalien
+- **Agenda**
+  ~~1. Sprint Update `00min`~~
+  2. Identify blockers / Answer Questions `30min`
+  3. This week's planning `15min`
+- **Reference:** 
+  - [Last week meeting minute](https://github.com/ipfs/testground/pull/69)
+  - [KanBan Board](https://app.zenhub.com/workspaces/test-ground-5db6a5bf7ca61c00014e2961/board)
+  - [Testing Infra Design](https://github.com/ipfs/testground/blob/master/docs/SPEC.md)
+  - [Testing Inventory](https://github.com/ipfs/testground/blob/master/docs/test-inventory.md)
+  - [Testing Improvement Tracker](https://docs.google.com/spreadsheets/d/1xyqyGUF-oe3x9ln88YonVeOMWWdknik74lVgL_3dBY8/edit#gid=0)
+
+:::
+
+## :mega: Update (Done, Shipped, Made Progress)
+> To be reviewed Async
+
+- ==@raulk==
+    - Little progress since I was on holiday last week, and now combating my backlog.
+    - Thought about multiversion test runs; planning to write a design blueprint.
+        - Gist: new concept "ensemble". An ensemble is a cohort of runs that run with the same ID but using different build outputs.
+        - Depends on tracking the dependency graph. Who can own this?
+    - Refactored the DHT plan. Currently a WIP PR; need to push it over the finish line this week.
+    - Arranging/negotiating travels for testground hack week.
+- ==@daviddias==
+    - Created Onboarding Work Plan for Henrique
+    - Coordinated the handing of the Infra Responsibilities from Jim to Anton
+    - Created the MVP/Primary Goal issue https://github.com/ipfs/testground/issues/196
+    - Made the plans and handled the logistics for the next Team "Onboarding" Hack Week
+    - Created a PRD for the TestGround Dashboard
+    - Reviewed PRs
+    - Did not have any tasty Thanksgiving Holidays food (we need that holiday here! :D)
+- ==@stebalien==
+    - Out Nov 27(Wed) - Nov 29(Fri) for Thanksgiving
+- ==@jimpick==
+    - Done:
+        - Hand off "Infra Lead" to Anton https://github.com/ipfs/testground/pull/192 (recording a Zoom call on Friday)
+    - Made progress:
+        - Started on Test Plan 4 - "Nodes Connectivity (Transports, Hole Punching, Relay)" https://github.com/ipfs/testground/issues/96
+        - Figured out how to implement a NAT inside a (privileged) Docker container - does the sidecar model work for this?
+        - Wrote up some starter ideas ... "connectivity obstacle course"
+    - Updated smlbench2 - as requested, rebased and renamed to data-transfer-datasets-random
+        - https://github.com/ipfs/testground/pull/74
+        - Not working yet - networking fails, needs further investigation
+        - Can't be merged yet ... it makes nasty modifications to Dockerfile.template in pkg/build/golang - needs to be cleaner
+        - Doesn't map well to proposed tests written for data-transfer-datasets-random
+    - PR: Add status badges to plans https://github.com/ipfs/testground/pull/200
+- ==@nonsense==
```

Comment:
@nonsense from https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments, did you get the chance to document how the daemon works?

Reply:
I added godoc to the daemon package, but this is not very discoverable, unless you use Go daily.

Probably architecture descriptions belong to the docs/SPEC.md , but it is a bit outdated, and I didn't want to take the task to update it right now.

I still have it on my todo list, and once the architecture solidifies, we could update altogether the docs/SPEC.md and/or add more sections to the README.md

I hope this makes sense.